book/src/ui-configuration.md

```diff
@@ -1,37 +1,109 @@
+    ```
+    BEACON_URL=http://localhost:5052
+    VALIDATOR_URL=http://localhost:5062
+    API_TOKEN=R6YhbDO6gKjNMydtZHcaCovFbQ0izq5Hk
+    SESSION_PASSWORD=your_password
+    ```
+
+ 1. You can now start Siren with:
+    ```bash
+    docker run --rm -ti --name siren -p 4443:443 --env-file $PWD/.env --net host sigp/siren
+    ```
+    If it fails to start, an error message will be shown. For example, the error
+    ```
+    http://localhost:5062 unreachable, check settings and connection
+    ```
+    means that the validator client is not running, or the `--http` flag is not provided. Another common error is:
+    ```
+    validator api issue, server response: 403
+    ```
+    which means that the API token is incorrect. Check that you have provided the correct token in the field `API_TOKEN` in `.env`.
+
+    When Siren is successfully run, you should see the log `LOG [NestApplication] Nest application successfully started +118ms`, indicating that Siren has started. 
+
+ 1. At the client computer, SSH to the server:
+    ```bash
+    ssh -L 3000:127.0.0.1:3000  username@server_IP
+    ```
+    You can now access siren by entering `localhost:3000` to a browser in the client computer. 
+
+
+Advanced users can mount their own certificates, see the `SSL Certificates` section below
+
+## Building From Source
+
+### Docker
+
+The docker image can be built with the following command:
+`docker build -f Dockerfile -t siren .`
+
+### Building locally
+
+To build from source, ensure that your system has `Node v18.18` and `yarn` installed.
+
+#### Build and run the backend
+
+Navigate to the backend directory `cd backend`. Install all required Node packages by running `yarn`. Once the installation is complete, compile the backend with `yarn build`. Deploy the backend in a production environment, `yarn start:production`. This ensures optimal performance.
+
+#### Build and run the frontend
+
+After initializing the backend, return to the root directory. Install all frontend dependencies by executing `yarn`. Build the frontend using `yarn build`. Start the frontend production server with `yarn start`.
+
+This will allow you to access siren at `http://localhost:3000` by default.
+
+## Advanced configuration
+
+### About self-signed SSL certificates
+
+By default, Siren will generate and use a self-signed certificate on startup.
+This will generate a security warning when you try to access the interface.
+We recommend to only disable SSL if you would access Siren over a local LAN or otherwise highly trusted or encrypted network (i.e. VPN).
+
+#### Generating persistent SSL certificates and installing them to your system
 
-> To access from another machine on the same network (192.168.0.200) set the Beacon Node and Validator Client `--http-address` as `192.168.0.200`. When this is set, the validator client requires the flag `--beacon-nodes http://192.168.0.200:5052` to connect to the beacon node.
```

Comment:
I am not sure about this part. I think this is not necessary if we can connect Siren via SSH. Proposing to remove this to make things simpler.

Reply:
The aim for Siren is to be as easily accessible as possible, asking them to expose a port over a reverse ssh tunnel is way more complicated. Just exposing the port and accessing the webapp directly that way fits better in the `end-user application`